### PR TITLE
feat: Add check changeset command

### PR DESCRIPTION
**Pull request by tylerbutler**
## Description

Adds a `flub check changesets` command which we will use in CI to determine whether a changeset was added. The existing changesets tools don't seem to work for us, hence this command.

I expanded the Repository class that contains our git logic to support checking for changed files, directories, packages, and release groups. This is not all used in this PR, but will be used in the future.

### CODE REVIEW

File: build-tools/packages/build-cli/package.json [hunk 1]
```diff
@@ -139,6 +139,7 @@
 		"moment": "^2.21.0",
 		"nyc": "^15.1.0",
 		"rimraf": "^4.4.0",
+		"ts-extras": "^0.11.0",
 		"ts-node": "^10.2.1",
```
Comment (tylerbutler) [inline build-tools/packages/build-cli/package.json:143]:
```suggestion
		"rimraf": "^4.4.0",
		"ts-node": "^10.2.1",
```

File: build-tools/packages/build-cli/src/lib/git.ts [hunk 2]
```diff
@@ -2,12 +2,17 @@
  * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
  * Licensed under the MIT License.
  */
+import { Context, Package } from "@fluidframework/build-tools";
+import path from "node:path";
+import readPkgUp from "read-pkg-up";
 import { SimpleGit, SimpleGitOptions, simpleGit } from "simple-git";
-// flase positive?
+
+// false positive?
```
Comment (sonalideshpandemsft) [inline build-tools/packages/build-cli/src/lib/git.ts:10]: nits: comment can be removed?

**Review by sonalideshpandemsft — APPROVED** [review]
lgtm

File: build-tools/packages/build-cli/src/lib/git.ts [hunk 3]
```diff
@@ -2,12 +2,17 @@
...
 // eslint-disable-next-line node/no-missing-import
```
Comment (tylerbutler) [inline build-tools/packages/build-cli/src/lib/git.ts:11]:
```suggestion

// type-fest seems to trigger this lint rule, which seems to be a false positive.
// eslint-disable-next-line node/no-missing-import
```

File: build-tools/packages/build-cli/src/lib/git.ts (hunk 2 above)
Comment (tylerbutler) [reply build-tools/packages/build-cli/src/lib/git.ts]: I expanded it to explain that the eslint disable shouldn't be needed but is for some reason.

**Pull request merged**